rita/src/rita_client/dashboard/interfaces.rs

```diff
@@ -0,0 +1,518 @@
+use rita_common::peer_listener::PeerListener;
+use rita_common::peer_listener::{Listen, UnListen};
+use settings::RitaCommonSettings;
+use KI;
+use SETTING;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct InterfaceToSet {
+    pub interface: String,
+    pub mode: InterfaceMode,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum InterfaceMode {
+    Mesh,
+    LAN,
+    WAN,
+    Unknown, // Ambiguous wireless modes like monitor or promiscuous
+}
+
+impl ToString for InterfaceMode {
+    fn to_string(&self) -> String {
+        match self {
+            InterfaceMode::Mesh => "mesh".to_owned(),
+            InterfaceMode::LAN => "LAN".to_owned(),
+            InterfaceMode::WAN => "WAN".to_owned(),
+            InterfaceMode::Unknown => "unknown".to_owned(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct GetInterfaces;
+
+impl Message for GetInterfaces {
+    type Result = Result<HashMap<String, InterfaceMode>, Error>;
+}
+
+impl Handler<GetInterfaces> for Dashboard {
+    type Result = Result<HashMap<String, InterfaceMode>, Error>;
+    fn handle(&mut self, _msg: GetInterfaces, _ctx: &mut Self::Context) -> Self::Result {
+        get_interfaces()
+    }
+}
+
+/// Gets a list of interfaces and their modes by parsing UCI
+pub fn get_interfaces() -> Result<HashMap<String, InterfaceMode>, Error> {
+    let mut retval = HashMap::new();
+
+    // Wired
+    for (setting_name, value) in KI.uci_show(Some("network"))? {
+        // Only non-loopback non-bridge interface names should get past
+        if setting_name.contains("ifname") && !value.contains("backhaul") && value != "lo" {
+            // it's a list and we need to handle that
+            if value.contains(",") {
+                for list_member in value.split(",") {
+                    retval.insert(
+                        list_member.replace(" ", "").to_string(),
+                        ethernet2mode(&value, &setting_name)?,
+                    );
+                }
+            } else {
+                retval.insert(value.clone(), ethernet2mode(&value, &setting_name)?);
+            }
+        }
+    }
+
+    // Wireless
+    for (setting_name, value) in KI.uci_show(Some("wireless"))? {
+        if setting_name.contains("ifname") {
+            retval.insert(value.clone(), wlan2mode(&value, &setting_name)?);
+        }
+    }
+
+    Ok(retval)
+}
+
+/// Find out a wired interface's mode (mesh, LAN, WAN) from the setting name
+pub fn ethernet2mode(ifname: &str, setting_name: &str) -> Result<InterfaceMode, Error> {
+    trace!(
+        "ethernet2mode: ifname {:?}, setting_name {:?}",
+        ifname,
+        setting_name
+    );
+
+    // Match parent section name
+    Ok(match &setting_name.replace(".ifname", "") {
+        s if s.contains("rita_") => InterfaceMode::Mesh,
+        s if s.contains("lan") => InterfaceMode::LAN,
+        s if s.contains("backhaul") => InterfaceMode::WAN,
+        other => bail!(
```

Comment:
This is very inconsiderate, I must've left it in by accident. This should ideally work the same as `wlan2mode` -  log via `warn!` and return `InterfaceMode::Unknown`